sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,654 @@
+            Note that this is supported only if all underlying estimators
+            support sample weights.
+
+        Returns
+        -------
+        self : object
+        """
+
+        self._validate_meta_estimator()
+
+        if self.estimators is None or len(self.estimators) == 0:
+            raise AttributeError(
+                "Invalid 'estimators' attribute, 'estimators' should be a list"
+                " of (string, estimator) tuples."
+            )
+
+        names, estimators_ = zip(*self.estimators)
+        self._validate_names(names)
+
+        has_estimator = False
+        for est in estimators_:
+            if est not in (None, 'drop'):
+                has_estimator = True
+
+        if not has_estimator:
+            raise ValueError(
+                "All estimators are None or 'drop'. At least one is required "
+                "to be an estimator."
+            )
+
+        if isinstance(self.predict_method, str):
+            if self.predict_method != 'auto':
+                raise AttributeError(
+                    "When 'predict_method' is a string, it should be 'auto'. "
+                    " Got {} instead.".format(self.predict_method)
+                )
+            predict_method = [self.predict_method] * len(estimators_)
+        else:
+            if len(self.estimators) != len(self.predict_method):
+                raise AttributeError(
+                    "When 'predict_method' is a list, it should be the same "
+                    "length as the list of estimators. Provided {} methods "
+                    "for {} estimators."
+                    .format(len(self.predict_method), len(self.estimators))
+                )
+            predict_method = self.predict_method
+
+        # Fit the base estimators on the whole training data. Those
+        # base estimators will be used in transform, predict, and
+        # predict_proba. They are exposed publicly.
+        self.estimators_ = Parallel(n_jobs=self.n_jobs)(
+            delayed(_parallel_fit_estimator)(clone(est), X, y, sample_weight)
+            for est in estimators_ if est not in (None, 'drop')
+        )
+
+        self.named_estimators_ = Bunch()
+        est_fitted_idx = 0
+        for name_est, org_est in zip(names, estimators_):
+            if org_est not in (None, 'drop'):
+                self.named_estimators_[name_est] = self.estimators_[
+                    est_fitted_idx]
+                est_fitted_idx += 1
+
+        # To train the meta-classifier using the most data as possible, we use
+        # a cross-validation to obtain the output of the stacked estimators.
+
+        # To ensure that the data provided to each estimator are the same, we
+        # need to set the random state of the cv if there is one and we need to
+        # take a copy.
+        random_state = check_random_state(self.random_state)
+        cv = check_cv(self.cv)
+        if hasattr(cv, 'random_state'):
```

Comment:
I think we only really need to do this if cv.random_state = None, in which case we don't need to have a separate `random_state` parameter here (note that the `check_cv(int)` `*KFold` instantiations are not randomised)... So the user could have provided their own `random_state` to the CV instance, or we can initialise with the global RandomState().

Reply:
Oh I see it makes sense to delegate to the cv object I think.